### PR TITLE
added EventManager & fixed bugs in GridLayoutManager

### PR DESCRIPTION
**Pull request by andy42**

### CODE REVIEW

File: src/main/java/ui/activity/Activity.java [hunk 1]
```diff
@@ -143,22 +131,35 @@ public Boolean dispatchKeyEvent (KeyEvent keyEvent){
...
-                return true;
-            } else if (isDragging && (motionEvent.getAction() == MotionEvent.ACTION_MOVE)) {
-                position = mousePos.sub(dragBarClickOffset);
-                return true;
+            if(isDragging && dragBarClickOffset == null){
+                dragBarClickOffset = Vector2Df.sub(motionEvent.getPosition(), this.position);
+            }
+            else if(isDragging){
+                position = motionEvent.getPosition().sub(dragBarClickOffset);
+            } else {
```
Comment (differentCharacter) [inline src/main/java/ui/activity/Activity.java:161]: you have `else if...` with bracket on the line above and `} else...` in the same if else statement xD

File: src/main/java/ui/view/LinearLayout.java [hunk 2]
```diff
@@ -200,7 +204,8 @@ public void onLayout (int width, int height, int x, int y) {
                 }
 
                 if ((this.gravity & Gravity.CENTER) == Gravity.CENTER) {
-                    childY = (int) ((height - totalLength) * 0.5f);
+                    //TODO: fix for when the totalLength is greater than height
+                    //childY = (int) ((height - totalLength) * 0.5f);
```
Comment (differentCharacter) [inline src/main/java/ui/view/LinearLayout.java:208]: Would it be better to just delete code that isn't in use and add a ticket to fix it?

File: src/main/java/ui/view/ViewGroup.java [hunk 3]
```diff
@@ -101,6 +101,7 @@ public void unFocus(){
         }
     }
 
+
```
Comment (differentCharacter) [inline src/main/java/ui/view/ViewGroup.java:104]: extra new line

File: src/main/java/ui/view/recyclerLayoutManager/GridLayoutManager.java [hunk 4]
```diff
@@ -24,7 +26,7 @@ public GridLayoutManager(int orientation, int columnCount){
         RecyclerView.ViewHolder lastChildAdded = null;
         float maxDimension = 0;
 
-        float columnWidth = (this.orientation == this.VERTICAL) ? recyclerView.dimensions.x/columnCount : recyclerView.dimensions.y/columnCount;
+        float columnWidth = (this.orientation == this.VERTICAL) ? recyclerView.getDimensions().x/columnCount : recyclerView.getDimensions().y/columnCount;
```
Comment (differentCharacter) [inline src/main/java/ui/view/recyclerLayoutManager/GridLayoutManager.java:29]: 😀

**Review by differentCharacter — APPROVED** [review]
Awesome, a few comments over trivial stuff, but loads of really good improvements here. Definitely a ✔️ from me.

File: src/main/java/ui/view/LinearLayout.java (hunk 2 above)
Comment (andy42) [reply src/main/java/ui/view/LinearLayout.java:208]: I still want the code, just needs to be added to. commented out as it brakes things as is

File: src/main/java/ui/view/ViewGroup.java (hunk 3 above)
Comment (andy42) [reply src/main/java/ui/view/ViewGroup.java:104]: deleted in net pull request

**Pull request merged**